api/v3/Payment.php

```diff
@@ -41,6 +41,9 @@
  *   Array of financial transactions which are payments, if error an array with an error id and error message
  */
 function civicrm_api3_payment_get($params) {
+  if (!CRM_Core_Permission::check('access CiviContribute')) {
```

Comment:
Also note that function will apply them depending on whether 'is_check_permission' is applied - you are applying unilaterally here.